server/repo-mgr.c

```diff
@@ -4424,3 +4425,35 @@ seaf_repo_manager_convert_repo_path (SeafRepoManager *mgr,
     return ret;
 }
 
+int
+seaf_repo_manager_set_repo_status(SeafRepoManager *mgr,
+                                  const char *repo_id, RepoStatus status)
+{
+    int ret = 0;
+
+    if (seaf_db_statement_query (mgr->seaf->db,
+                                 "UPDATE RepoInfo SET status=? "
+                                 "WHERE repo_id=? OR repo_id IN "
+                                 "(SELECT repo_id FROM VirtualRepo WHERE origin_repo=?)",
+                                 3, "int", status,
+                                 "string", repo_id, "string", repo_id) < 0)
+        ret = -1;
+
+    return ret;
+}
+
+int
+seaf_repo_manager_get_repo_status(SeafRepoManager *mgr,
+                                  const char *repo_id)
+{
+    char *sql;
+    gboolean is_vir = seaf_repo_manager_is_virtual_repo(mgr, repo_id);
```

Comment:
It should be sufficient to directly get the status of the virtual repo since we'll disallow creating virtual repos if a repo is in read-only status.

BTW, you also have to disallow creating virtual repos from read-only repos.